pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +308,32 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, registryConfigPath) {
+			key := keyParts[len(keyParts)-1]
+			index, err := strconv.Atoi(string(question.Variable[len(registryConfigPath)+1]))
+			if err != nil {
+				return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
+			}
+			if len(existingRegistries) > index {
+				existingRegistries[index][key] = val
+			} else {
+				existingRegistries = append(existingRegistries, map[string]interface{}{key: val})
+			}
+		} else {
+			values.PutValue(dataFromTemplate, val, keyParts...)
+		}
 
 		questionMap, err := convert.EncodeToMap(question)
 		if err != nil {
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error reading clusterTemplate questions")
 		}
 		revisionQuestions = append(revisionQuestions, questionMap)
 	}
+	if len(existingRegistries) > 0 {
+		existingRegistries[0][managementv3.PrivateRegistryFieldIsDefault] = interface{}(true)
```

Comment:
Is the cast here necessary?
```suggestion
		existingRegistries[0][managementv3.PrivateRegistryFieldIsDefault] = true
```